### PR TITLE
Add optimization options to Lagom CMakelists

### PR DESCRIPTION
**Pull request by TheCrott**
Some bugs can be found with certain optimization options. In my previous experience of fuzzing, I play with different optimization flags.
This is the example crash in `FuzzASN1` found using only `-O0` and `integer` sanitizer https://github.com/SerenityOS/serenity/issues/10165
cc @bgianfo

### CODE REVIEW

File: Meta/Lagom/CMakeLists.txt [hunk 1]
```diff
@@ -63,7 +63,17 @@ endif()
...
+if (OPTIMIZATION_FLAG STREQUAL "O0")
+    add_compile_options(-O0)
+elseif (OPTIMIZATION_FLAG STREQUAL "O1")
+    add_compile_options(-O1)
+elseif (OPTIMIZATION_FLAG STREQUAL "O2")
+    add_compile_options(-O2)
+elseif (OPTIMIZATION_FLAG STREQUAL "O3")
+    add_compile_options(-O3)
+else ()
+    add_compile_options(-O2)
```
Comment (ADKaster) [inline Meta/Lagom/CMakeLists.txt:75]: Isn't setting these flags what `CMAKE_BUILD_TYPE` does already? https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html

Comment (kvuj) [reply Meta/Lagom/CMakeLists.txt:75]: https://github.com/SerenityOS/serenity/blob/master/CMakeLists.txt#L7

Comment (Hendiadyoin1) [reply Meta/Lagom/CMakeLists.txt:75]: anyway, there is a smarter way to do this,
you can just asign to the variable if it is not set or give it a standard value as an option
and then just add the string contained to the compile options (maybe even just the number), maybe with some error checking before hand, to ensure that it is a valid optimization level

**Comment by friendlyanon** [comment]
This is unidiomatic CMake, you are also hardcoding platform specific details.

You have [CMAKE\_&lt;LANG&gt;\_FLAGS\_&lt;CONFIG&gt;](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS_CONFIG.html) to define these options for whatever language to whatever optimization level from the command line (or a preset in the future).

**Pull request closed without merging**